tests/chainerx_tests/unit_tests/routines_tests/test_logic.py

```diff
@@ -49,32 +53,41 @@ def _to_array_safe(xp, a_object, dtype):
     ([0], [[0, 1, 2], [3, 4, 5]]),
     ([[0], [1]], [0, 1, 2]),
 ])
-@pytest.mark.parametrize('cmp_op, chx_cmp, np_cmp', [
-    (lambda a, b: a == b, chainerx.equal, numpy.equal),
-    (lambda a, b: a != b, chainerx.not_equal, numpy.not_equal),
-    (lambda a, b: a > b, chainerx.greater, numpy.greater),
-    (lambda a, b: a >= b, chainerx.greater_equal, numpy.greater_equal),
-    (lambda a, b: a < b, chainerx.less, numpy.less),
-    (lambda a, b: a <= b, chainerx.less_equal, numpy.less_equal),
+@chainer.testing.parameterize_pytest('cmp_op,module_func', [
+    (lambda a, b: a == b, 'equal'),
+    (lambda a, b: a != b, 'not_equal'),
+    (lambda a, b: a > b, 'greater'),
+    (lambda a, b: a >= b, 'greater_equal'),
+    (lambda a, b: a < b, 'less'),
+    (lambda a, b: a <= b, 'less_equal'),
 ])
-@pytest.mark.parametrize_device(['native:0', 'cuda:0'])
-def test_cmp(device, cmp_op, chx_cmp, np_cmp, a_object, b_object, dtype):
-    a_np = _to_array_safe(numpy, a_object, dtype)
-    b_np = _to_array_safe(numpy, b_object, dtype)
-    if a_np is None or b_np is None:
-        return
-
-    a_chx = chainerx.array(a_np)
-    b_chx = chainerx.array(b_np)
-
-    chainerx.testing.assert_array_equal_ex(
-        cmp_op(a_chx, b_chx), cmp_op(a_np, b_np))
-    chainerx.testing.assert_array_equal_ex(
-        cmp_op(b_chx, a_chx), cmp_op(b_np, a_np))
-    chainerx.testing.assert_array_equal_ex(
-        chx_cmp(a_chx, b_chx), np_cmp(a_np, b_np))
-    chainerx.testing.assert_array_equal_ex(
-        chx_cmp(b_chx, a_chx), np_cmp(b_np, a_np))
+@pytest.mark.filterwarnings('ignore:invalid value encountered in ')
```

Comment:
Could you leave a comment about in what cases these warnings are raised?

Reply:
Fixed the comment position.